colossalai/pipeline/schedule/_utils.py

```diff
@@ -118,15 +149,18 @@ def merge_batch(data: List[Any]) -> Any:
     flattened_data = []
     tree_spec = None
     for d in data:
-        elems, tree_spec = tree_flatten(d)
+        # elems should be an instance of OrderedDict
+        elems, tree_spec = tree_flatten_hf(d)
         flattened_data.append(elems)
     merged_data = []
-    for elem_batch in zip(*flattened_data):
+
+    for elem_batch in zip(*(flattened_data)):
```

Comment:
What's the effect of the brackets?

Reply:
actually no, it is for other testing and I will remove it